core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -2173,9 +2175,15 @@ public ResolvedFunction resolve(Session session, FunctionBinding functionBinding
     @Override
     public boolean isAggregationFunction(Session session, QualifiedName name)
     {
-        return functions.getFunctions(name).stream()
-                .map(FunctionMetadata::getKind)
-                .anyMatch(AGGREGATE::equals);
+        return functionResolver.isAggregationFunction(session, name, this::getFunctions);
+    }
+
+    private Collection<FunctionMetadata> getFunctions(CatalogSchemaFunctionName name)
+    {
+        if (name.getCatalogName().equals(GLOBAL_CATALOG)) {
```

Comment:
Do we prevent creating a catalog with this name?

Reply:
The "system" catalog is the built in catalog that is information like the nodes in the cluster, and metadata for the jdbc driver.